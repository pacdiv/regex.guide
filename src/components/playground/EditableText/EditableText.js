import PropTypes from "prop-types"
import React, { Component, createRef } from "react"

import { EditableTextContainer, MenuButton } from "./EditableText.style"
import editIcon from "../../../images/icon-edit.png"

class ConditionSentence extends Component {
  static propTypes = {
    onClick: PropTypes.func,
    sentence: PropTypes.string
  }

  containerRef = createRef()

  onButtonClick = () => this.props.onClick(this.containerRef)

  render() {
    return (
      <EditableTextContainer ref={this.containerRef}>
        {this.props.sentence}
        <MenuButton onClick={this.onButtonClick} type="button">
          <img alt="menu" src={editIcon} />
        </MenuButton>
      </EditableTextContainer>
    )
  }
}

export default ConditionSentence
