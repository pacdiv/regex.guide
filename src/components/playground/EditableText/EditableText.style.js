import styled from "@emotion/styled"

import { Button } from "../../utils"

export const EditableTextContainer = styled.div`
  align-items: center;
  display: flex;
  justify-content: center;
  margin: 0.75em 0 0;
  position: relative;

  &:last-of-type {
    margin-bottom: 0.75em;
  }

  &:only-child {
    margin: 0;
  }
`

export const MenuButton = styled(Button)`
  background-color: #ffffff;
  height: 1em;
  line-height: 1px;
  margin: 0;
  width: 1em;

  img {
    margin: 0;
    max-width: 1em;
  }
`
