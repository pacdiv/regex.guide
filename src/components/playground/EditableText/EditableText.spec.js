import React from "react"
import { fireEvent, render } from "@testing-library/react"
import "@testing-library/jest-dom/extend-expect"

import EditableText from "./EditableText"

test("EditableText component", async () => {
  const onClick = jest.fn()
  const { container, getByAltText } = render(
    <EditableText
      label="demo"
      onClick={onClick}
      sentence="Is this a placeholder?"
    />
  )

  expect(container).toMatchSnapshot()
  fireEvent.click(getByAltText("menu").closest("button"))
  expect(onClick).toHaveBeenCalledTimes(1)
})
