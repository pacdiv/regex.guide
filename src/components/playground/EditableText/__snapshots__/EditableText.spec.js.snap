// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditableText component 1`] = `
<div>
  <div
    class="css-134iufa"
  >
    Is this a placeholder?
    <button
      aria-label="demo-editable-text-button"
      class="css-tija8u"
      type="button"
    >
      <img
        alt="menu"
        src="test-file-stub"
      />
    </button>
  </div>
</div>
`;
